app.rb

```diff
@@ -0,0 +1,171 @@
+require './book'
+require './classroom'
+require './nameable'
+require './rental'
+require './teacher'
+require './student'
+
+class App
+  def start
+    puts 'Welcome to School Library App!'
+    puts # blank
+    @choice = ''
+    options
+  end
+
+  def options
+    puts 'Please choose an option by enterin a number:'
+    puts '1 - List all books', '2 - List all people', '3 - Create a person', '4 - Create a book',
+         '5 - Create a rental', '6 - List all rentals for a given person id', '7 - Exit'
+    @choice = gets.chomp
+
+    if @choice.upcase == '7'
+      puts 'Goodbye!'
+    elsif @choice.to_i != 0 && @choice.to_i.between?(1, 6)
+      sleep(1)
+      enter_menu
+    else
+      puts 'Please choose a valid option!'
+      puts # blank
+    end
+  end
+
+  def enter_menu
+    case @choice
+    when '1' then list_all_books
+    when '2' then list_all_people
+    when '3' then create_person
+    when '4' then create_book
+    when '5' then create_rental
+    when '6' then list_rental_by_id
+    end
+  end
+
+  def list_all_books
+    books = Book.all
+    if books.empty?
+      puts 'There are no books added yet.'
+    else
+      puts 'These are all the books:'
+      books.each do |book|
+        puts "\"#{book.title}\" by #{book.author}"
+      end
+    end
+    puts # blank
+  end
+
+  def list_all_people
+    students = Student.all
+    teachers = Teacher.all
+
+    if students.empty? && teachers.empty?
+      puts 'There are no people added yet.'
+    else
+      puts 'These are all the people:'
+      students.each do |student|
+        puts "#{student.name}, Student aged #{student.age} with ID: #{student.id}."
+      end
+      teachers.each do |teacher|
+        puts "#{teacher.name}, #{teacher.specialization} Teacher aged #{teacher.age} with ID: #{teacher.id}."
+      end
+    end
+    puts # blank
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [input the number]: '
+    input = gets.chomp
+    case input
+    when '1' then create_student
+
+    when '2' then create_teacher
+    end
+  end
+
+  def create_student
+    puts 'Please add details of Student...'
+    print 'Name: '
+    s_name = gets.chomp
+    print 'Age: '
+    s_age = gets.chomp.to_i
+    print 'Has parent permission (Y/N): '
+    s_perm = gets.chomp.upcase == 'Y'
+    Student.new(s_age, s_perm, s_name).make_student
+  end
+
+  def create_teacher
+    puts 'Please add details of Teacher...'
+    print 'Name: '
+    t_name = gets.chomp
+    print 'Age: '
+    t_age = gets.chomp.to_i
+    print 'Specialization: '
+    t_spec = gets.chomp
+    Teacher.new(t_spec, t_age, t_name)
+  end
```

Comment:
- [ ] [REQUIRED] Hello @obed300 Great effort trying to fix the recommended changes by the previous reviewer, however, I can not create a person. The app exits automatically in the process. Kindly take a look below. Let's fix this to ensure the application works as expected 💪🏽 
<img width="878" alt="Screenshot 2023-05-06 at 20 36 54" src="https://user-images.githubusercontent.com/58692590/236643773-f1fc5d08-3130-4b2d-9680-a8238dadcbe6.png">